utils/s3/client.cc

```diff
@@ -770,7 +781,16 @@ class client::upload_sink final : public client::upload_sink_base {
             co_await upload_part(std::move(_bufs));
         }
         if (upload_started()) {
-            co_await finalize_upload();
```

Comment:
Same code in upload_jumbo_sink needs same update

Reply:
not exactly the same, I think it should use close instead of calling abort directly
```
if (upload_started()) {
    std::exception_ptr ex;
    try {
        co_await finalize_upload();
    } catch (...) {
        ex = std::current_exception();
    }
    if (ex) {
        co_await close();
        std::rethrow_exception(ex);
    }
}
```
WDYT?

Reply:
Why close()? It's the same .flush() method that shouldn't close itself on failure

Reply:
you are right, for some reason I was under impression the `abort` then will be called twice, once here and the second one in the final `close`, but all good